### ISSUE TITLE
Can't update to 1.6.6 on Windows

### ISSUE THREAD

**Issue by nickautomatic** [bower/bower#2044]
After running `npm install -g bower` to update to 1.6.6, the install process ends with:

    npm ERR! code 1

Bower is then broken: if I run, eg. `bower -v` I get:

```
module.js:339
    throw err;
    ^

Error: Cannot find module 'js-yaml'
    at Function.Module._resolveFilename (module.js:337:15)
    at Function.Module._load (module.js:287:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (c:\Users\Nick F\AppData\Roaming\npm\node_modules\bower\node_modules\configstore\index.js:9:12)
    at Module._compile (module.js:435:26)
    at Object.Module._extensions..js (module.js:442:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
```

I then ran `npm install -g bower@1.6.5` to get the old version back, which got everything working again.

**Comment by pertrai1** [comment]
@nickautomatic What version of Windows are you running?

**Comment by nickautomatic** [comment]
@pertrai1 Windows 8.1 Pro. I'm running bower from bash (the version that comes with [Git for Windows](https://git-for-windows.github.io/))

**Comment by nickautomatic** [comment]
One possibility that occurred to me: my username has a space in it - I know that's something that has caused problems with npm in the past (I think it's something that is allowed in Windows but not in Unix-like operating systems... I might be wrong there though!)

**Comment by pertrai1** [comment]
@nickautomatic that could be the issue. I tried to reproduce on my Windows machine and all installed properly. I did notice that js-yaml is one of the last packages to be installed for Bower, so I was thinking that it might have just been an issue with getting js-yaml. Not sure if this would be the case or not. Maybe try to run again and see if you get the same result?

**Comment by nickautomatic** [comment]
@pertrai1 Ok, I tried again, but get the exact same problem I'm afraid...

**Comment by tokyoincode** [comment]
Happens on El Capitan (OS X) as well. However, 1.6.6 installs fine on Yosemite.

**Comment by tokyoincode** [comment]
Try uninstalling Bower and then reinstalling it. Might work for you as well.
(Credit goes to my boss)

**Comment by riyadhalnur** [comment]
Confirmed not working in El Capitan across multiple Node versions

**Comment by rscnt** [comment]
@riyadhalnur, @nickautomatic which version of npm are you using?

**Comment by riyadhalnur** [comment]
i'm on 3.5.0

**Comment by riyadhalnur** [comment]
@nickautomatic 1.6.7 was just pushed out a while ago. Can you please check if that fixes things for you? The latest version is good on OSX

**Comment by nickautomatic** [comment]
@rscnt, @riyadhalnur: Ah, sorry, I won't be able to check until Monday now, but I will try it then and let you know. Re: npm version - again, can't check the exact version, but I know I've updated npm fairly recently so it would be 3 point something.

For what it's worth, I've just tried installing 1.6.7 on the machine I'm currently using (also Windows 8.1) without running into this issue so possibly 1.6.7 fixes the issue (though on this machine my username doesn't have a space in it, so obviously it's not a perfect test). I'll update to 1.6.7 on my work machine on Monday and let you know how it goes.
